src/base/utils/fs.cpp

```diff
@@ -99,9 +99,13 @@ QString Utils::Fs::folderName(const QString &filePath)
 }
 
 /**
- * This function will first remove system cache files, e.g. `Thumbs.db`,
- * `.DS_Store`. Then will try to remove the whole tree if the tree consist
- * only of folders
+ * This function will first check if there are only system cache files, e.g. `Thumbs.db`,
+ * `.DS_Store` and/or only temp files that end with '~', e.g. `filename~`.
+ * If they are the only files it will try to remove them and delete the folder.
+ * This action will be performed for each subfolder starting from the deepest folder.
+ * There is an inherent race condition here. A file might appear after it is checked
```

Comment:
Is the race condition worth mentioning? Nearly all FS operation has race condition.